cdap-app-fabric/src/test/java/co/cask/cdap/internal/AppFabricTestHelper.java

```diff
@@ -147,7 +148,11 @@ public static void ensureNamespaceExists(Id.Namespace namespace) throws Exceptio
   public static void ensureNamespaceExists(Id.Namespace namespace, CConfiguration cConf) throws Exception {
     NamespaceAdmin namespaceAdmin = getInjector(cConf).getInstance(NamespaceAdmin.class);
     if (!namespaceAdmin.exists(namespace)) {
-      namespaceAdmin.create(new NamespaceMeta.Builder().setName(namespace).build());
+      try {
+        namespaceAdmin.create(new NamespaceMeta.Builder().setName(namespace).build());
+      } catch (NamespaceAlreadyExistsException e) {
```

Comment:
this is just to protect against race condition?
What motivated you to put this here? were you running into the race condition?